frontend/packages/console-plugin-sdk/src/typings/overview.ts

```diff
@@ -18,7 +19,10 @@ namespace ExtensionProperties {
     required: string;
 
     /** util to check get resources. */
-    utils: (dc: K8sResourceKind, props: OverviewMainContentProps) => ResourceItem | undefined;
+    utils: (
+      dc: K8sResourceKind,
+      props: OverviewMainContentProps,
+    ) => ResourceItem | undefined | { pipelines: PipelineOverviewItem };
```

Comment:
Not that it'll help clean this up, but any reason why we are returning an object with only 1 key? Why not just the value (type in this case)?

Reply:
This method will execute here `element` https://github.com/openshift/console/pull/3014/files#diff-0dffb9abd12f8591b1d227efedf17dcbR608 and it uses the spread operator. Returning a value like this will add `pipelines` prop to `overviewItem`.

Reply:
Something about changing this type to solve our problem feels like a mis-use of this code. All the other utilities are able to send back a map to `K8sResourceKind[]`, I know we have two types here... perhaps we need to produce two keys... `pipelines` and `pipelineRuns`. Instead of wrapping them up in their own (`PipelineOverviewItem`)

Reply:
Agreed! After creating this utility and using through a plugin, Even I'm in favor of creating two separate keys instead of wrapping them up in their own.